test/rekt/features/broker/data_plane.go

```diff
@@ -407,3 +414,440 @@ func brokerEventVersionNotUpgraded(ctx context.Context, t feature.T) {
 
 	// Create a trigger,
 }
+
+func deliveredCESpecs(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	uuids := map[string]string{
+		uuid.New().String(): "1.0",
+		uuid.New().String(): "0.3",
+	}
+
+	for id, version := range uuids {
+		source := feature.MakeRandomK8sName("source")
+		sink := feature.MakeRandomK8sName("sink")
+		triggerName := feature.MakeRandomK8sName("trigger")
+		event := test.FullEvent()
+		event.SetID(id)
+		event.SetSpecVersion(version)
+
+		// Creating sink
+		eventshub.Install(sink, eventshub.StartReceiver)(ctx, t)
+
+		// Point the Trigger subscriber to the sink svc.
+		cfg := []manifest.CfgFn{trigger.WithSubscriber(service.AsKReference(sink), "")}
+
+		// Creating trigger
+		trigger.Install(triggerName, brokerName, cfg...)
+
+		eventshub.Install(source,
+			eventshub.StartSenderToResource(broker.GVR(), brokerName),
+			eventshub.InputEvent(event),
+		)
+
+		assert.OnStore(sink).MatchEvent(test.HasId(event.ID()), test.HasSpecVersion(version)).Exact(1)
```

Comment:
is it better to do 
`assert.OnStore(sink).MatchEvent(test.HasId(event.ID()), test.HasSpecVersion(version)).Exact(1)(ctx, t)`
or 
```
store := eventshub.StoreFromContext(ctx, source)
events := knconf.Correlate(store.AssertAtLeast(t, 2, knconf.SentEventMatcher("")))
for _, e := range events {
	if e.Response.StatusCode < 200 || e.Response.StatusCode > 299 {
		t.Errorf("Expected statuscode 2XX for sequence %d got %d", e.Response.Sequence, e.Response.StatusCode)
	}
}
```

Reply:
@creydr For now, I have done what was suggested in this comment at every place because I observed that we are using event assert in recent commits and `eventshub.StoreFromContext` in older ones.

I can definitely use the latter if i should